src/components/App.js

```diff
@@ -30,9 +30,90 @@ const parseOptions = options => {
     .filter(str => str !== '')
 }
 
+/**
+ * getOptionsFromQueryParams :: string
+ * 
+ * Allows sharing URLs such as ?b=idle,loading,bad,ok
+ */
+const getOptionsFromQueryParams = () => {
+  if (typeof URLSearchParams !== 'undefined') {
+    const params = new URLSearchParams(document.location.search.substring(1))
+    return params.get('options') || ''
+  }
+
+  return ''
+}
+
+/**
+ * syncLocation :: string -> void
+ * 
+ * Replaces the browser window's current history state with updated options.
+ * The location is replaced while the user is typing and pushed when they add
+ * a new boolean or completely replace the previous options with a string
+ * that doesn't match the current set of options.
+ * 
+ * @param {string} options Comma separated list of options
+ */
+const syncLocation = (options) => {
+  if (!options.trim()) {
+    return
+  }
+
+  const current = getOptionsFromQueryParams()
+
+  // options minus trailing comma
+  const next = options.replace(/,$/, '')
+
+  // If the next string contains more information than the previous location
+  if (options.length > current.length) {
+    // If the user has added a comma or completely replaced
+    if (options[options.length - 1] === ',' || (new RegExp(current)).test(options) === false) {
+      history.pushState({}, '', `?options=${next}`)
+    }
+  } else {
+    // If the user replaced the options string with a shorter string, e.g. copy/paste
```

Comment:
This comment is inaccurate, the lengths could be the same and reach this branch.

Also, is worrying about pushState important? Is there any concern with browser history here? I can't imagine a user using back and forth history while using the app when they can type it faster.

It's probably sufficient just to update the query params as someone types so they have a shareable link.